packages/agents/cartographer/poller/src/lib/operations/routers.ts

```diff
@@ -35,6 +35,31 @@ export const updateRouters = async () => {
   }
 };
 
+export const updateAssets = async () => {
+  const {
+    adapters: { subgraph, database },
+    logger,
+    domains,
+  } = getContext();
+  const { requestContext, methodContext } = createLoggingContext(updateAssets.name);
+
+  const assets: Asset[] = [];
+  for (const domain of domains) {
+    const offset = await database.getCheckPoint("asset_" + domain);
+    const limit = 100;
+    logger.debug("Retrieving assets", requestContext, methodContext, {
+      domain: domain,
+      offset: offset,
+      limit: limit,
+    });
+
+    const _assets = await subgraph.getAssets(domain);
+    assets.push(..._assets);
+  }
+  await database.saveAssets(assets);
```

Comment:
There's no checkpoint here. The number of assets is small so we can just query them all and we don't need any type of cursor since there's no status changes.